src/services/zip-tax.ts

```diff
@@ -73,7 +72,20 @@ export default class ZipTaxService extends AbstractTaxService {
   }
 
   private buildAddressString(address: Address) {
-    return `${address.address_1} ${address.address_2}, ${address.city} ${address.province}, ${address.postal_code}`;
+    return Object.entries(address).reduce(
+      (prev, [key, value], index, array) => {
+        if (index === array.length - 1) return `${prev}${value}`;
```

Comment:
let's add some helper `const`s to make this easier to read,
```
const isLastEntry = index === array.length -1;
const text = value ?? '';

if (isLastEntry) return `${prev}${text}`;

const [nextKey, nextValue] = array[index + 1];
const addSpace = !!nextValue ? ' ' : '';
const addComma = ['city', 'country_code'].includes(nextKey)
          ? ','
          : '';

 return `${prev}${text}${addComma}${addSpace}`

```

Reply:
this is resolved!